docs/guide/rolldown.md

```diff
@@ -92,9 +110,34 @@ The `rolldown-vite` package is a temporary solution to gather feedback and stabi
 
 We encourage you to try out `rolldown-vite` and contribute to its development through feedback and issue reports.
 
+In the future, we will also introduce a "Full Bundle Mode" for Vite, which will server bundled files in production _and development mode_.
+
+### Why introducing a Full Bundle Mode?
+
+Vite is known for its unbundled dev server approach, which is a main reason for Vite's speed and popularity when it was first introduced. This approach was initially an experiment to see just how far we could push the boundaries of development server performance without traditional bundling.
+
+However, as projects scale in size and complexity, two main challenges have emerged:
+
+1. **Development/Production inconsistency**: The unbundled JavaScript served in development versus the bundled production build creates different runtime behaviors. This can lead to issues that only manifest in production, making debugging more difficult.
+
+2. **Performance degradation on page refresh**: When running the development server for large applications, a hard page refresh can trigger hundreds or even thousands of network requests as each module is loaded individually. While this is somewhat mitigated by HTTP/2 multiplexing, it still creates overhead that impacts startup performance, especially on slower networks or devices.
```

Comment:
Mentioning "slower networks / devices" is somewhat confusing. The wording here sounds like talking about production, not the dev server. We need to be explicit that the number of requests is impacting dev server startup and page refresh.

Some other pounts:

- The network bottleneck becomes more noticeable if the developers access the dev server through a network proxy (which happens in some setups).

- To our testing, HTTP/2 doesn't help much in terms of local page loading speed, probably due to the max concurrent requests that can be handled by the dev server.

Reply:
LGTM after this is addressed.

Reply:
Updated the part removing HTTP/2 and reflecting the proxy part + reinforcing that this is a dev issue & impacts page refresh as well as dev server startup.